### PR TITLE
fix some printing of special characters

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes https://github.com/JuliaLang/TOML.jl/issues/22

cc @c42f

### CODE REVIEW

File: src/print.jl [hunk 1]
```diff
@@ -1,9 +1,44 @@
...
+            Base.print(io, '\\', '"')
+        elseif c == '\\'
+            Base.print(io, "\\", '\\')
+        elseif Base.iscntrl(c)
+            Base.print(io, "\\u")
+            @static if is_printf_bugged
+                Base.print(io, lpad(UInt32(c), 4, '0'))
+            else
+                Printf.@printf(io, "%.4i", UInt32(c))
+            end
```
Comment (c42f) [inline src/print.jl:35]: Should be hex?

```suggestion
            Base.print(io, string(UInt32(c), base=16, pad=4))
```

(Or use `Printf.@printf(io, "%.4x", UInt32(c))` if that's faster... both allocate a `StringVector` internally though, so I guess they're about the same)

File: src/print.jl [hunk 2]
```diff
@@ -1,9 +1,44 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 import Dates
+import Printf
 
 import ..isvalid_barekey_char
 
+const is_printf_bugged = Printf.@sprintf("%.4i", 1) == "1"
+
```
Comment (c42f) [inline src/print.jl:9]:
```suggestion
```

File: test/print.jl [hunk 3]
```diff
@@ -58,3 +58,13 @@ end
     d = TOML.parse(s)
     @test toml_str(d) == "user = \"me\"\n\n[julia]\n\n[option]\n"    
 end
+
+@testset "special characters" begin
+    s = """
+    "\U1f355 \0 \x0 \x1 \t \b" = "\U1f355 \0 \x0 \x1 \t \b"
+    """
+    @test roundtrip(s)
+
```
Comment (c42f) [inline test/print.jl:67]: Here's a test which the current code is failing.

```suggestion

    @test sprint(TOML.print, Dict("\x7f"=>1)) = "\"\\u007f\" = 1\n"
```

**Review by c42f — COMMENTED** [review]
Cool, thanks.  Looks like there's a bug with hex vs decimal.  I added some suggestions.

File: src/print.jl (hunk 1 above)
Comment (KristofferC) [reply src/print.jl]: :+1:

**Comment by KristofferC** [comment]
Thanks for the help @c42f

**Pull request merged**